pkg/alertmanager/operator.go

```diff
@@ -881,22 +882,22 @@ func (c *Operator) selectAlertManagerConfigs(ctx context.Context, am *monitoring
 					}
 				}
 
-				if pdConfig.HTTPConfig == nil {
-					continue
+				if pdConfig.HTTPConfig != nil {
+					configureHTTPConfigInStore(ctx, store, pdConfig.HTTPConfig, amcNamespace, pdcKey)
 				}
+			}
+
+			for j, ogConfig := range receiver.OpsGenieConfigs {
```

Comment:
Alertmanager requires that at least one of `id`, `name` or `username` is not empty. The value of `type` should also match `^(team|user|escalation|schedule)$`. We need to enforce it here.

Reply:
Would the prefered behavior be to skip the responder that is not matching here or the whole opsgenie config?

Reply:
I would skip the whole config.

Reply:
Thanks. The latest commit should address your comment :D